rice-middleware/impl/src/main/java/org/kuali/rice/kim/service/impl/UiDocumentServiceImpl.java

```diff
@@ -2648,52 +2648,66 @@ protected void convertCheckboxAttributeData(RoleMemberAttributeDataBo roleMember
 	}
 
 	protected List<DelegateMemberBo> getDelegationMembers(List<RoleDocumentDelegationMember> delegationMembers,
-			List<DelegateMemberBo> origDelegationMembers, List<DelegateMemberBo> allOrigMembers, boolean activatingInactive, String newDelegationIdAssigned){
+			List<DelegateMemberBo> origDelegationMembers, List<DelegateMemberBo> allOrigMembers,
+			boolean activatingInactive, String newDelegationIdAssigned) {
 		List<DelegateMemberBo> delegationsMembersList = new ArrayList<DelegateMemberBo>();
 		DelegateMemberBo newDelegationMemberImpl;
 		DelegateMemberBo origDelegationMemberImplTemp = null;
 		List<DelegateMemberAttributeDataBo> origAttributes;
 		String delegationMemberId = "";
-		if(CollectionUtils.isNotEmpty(delegationMembers)){
-			for(RoleDocumentDelegationMember delegationMember: delegationMembers){
+		if (CollectionUtils.isNotEmpty(delegationMembers)) {
+			for (RoleDocumentDelegationMember delegationMember : delegationMembers) {
 				newDelegationMemberImpl = new DelegateMemberBo();
 				KimCommonUtilsInternal.copyProperties(newDelegationMemberImpl, delegationMember);
-                newDelegationMemberImpl.setType(MemberType.fromCode(delegationMember.getMemberTypeCode()));
-				if(ObjectUtils.isNotNull(origDelegationMembers)){
-					for(DelegateMemberBo origDelegationMember: origDelegationMembers){
-						if(activatingInactive && StringUtils.equals(origDelegationMember.getMemberId(), newDelegationMemberImpl.getMemberId()) &&
-								StringUtils.equals(newDelegationMemberImpl.getDelegationId(), newDelegationIdAssigned) &&
-								!origDelegationMember.isActive(getDateTimeService().getCurrentTimestamp())){
+

```

Comment:
I added the following comment and two lines afterwards to set the versionNumber and objectId to null.